libraries/Microsoft.Bot.Connector/Authentication/JwtTokenValidation.cs

```diff
@@ -62,22 +62,28 @@ public static async Task<ClaimsIdentity> AuthenticateRequest(IActivity activity,
             if (string.IsNullOrWhiteSpace(authHeader))
             {
                 var isAuthDisabled = await credentials.IsAuthenticationDisabledAsync().ConfigureAwait(false);
-                if (isAuthDisabled)
+                if (!isAuthDisabled)
                 {
-                    // In the scenario where Auth is disabled, we still want to have the
-                    // IsAuthenticated flag set in the ClaimsIdentity. To do this requires
-                    // adding in an empty claim.
-                    return new ClaimsIdentity(new List<Claim>(), "anonymous");
+                    // No Auth Header and Auth is required. Request is not authorized.
+                    throw new UnauthorizedAccessException();
                 }
 
-                // No Auth Header. Auth is required. Request is not authorized.
-                throw new UnauthorizedAccessException();
+                // Check if the activity is for a skill call and is coming from the Emulator.
+                if (activity.ChannelId == Channels.Emulator && activity is Activity act && act.RelatesTo != null)
+                {
+                    // Return an anonymous claim with an anonymous skill AppId
+                    return new ClaimsIdentity(new List<Claim>() { new Claim(AuthenticationConstants.AppIdClaim, AuthenticationConstants.AnonymousSkillAppId) }, "anonymous");
```

Comment:
consider: Should we define a method that creates this 'un-authenticated claims identity? We have this exact code in one more place. i.e. return SkillsValidation.AnonymousIdentity which is a constant.

Reply:
I am not sure, this is used in Microsoft.Bot.Builder and Microsoft.Bot.Connector, so we will need to add another static method on SkillValidation (or some other class) and make it public. Given that we are not completely sure this is the right approach, I'd rather leave with some internal duplication and not add a new public method that we will need to deprecate if we change our mind.
Thoughts?

Reply:
I get that point. However now we went from places to 3 places. You cannot guarantee that a future dev will remember to update all 3 places if they change it. Furthermore, we cannot even guarantee that a code reviewer will remember that there are 3 places that need to be updated. 

If you wnat to keep the internal dupe until we prove it, that's fair, but eventually would love to converge this. LEaving it at your criteria.

Reply:
Added SkillValidation.CreateAnonymousSkillClaim() helper and tests and updated related code.